test/DefaultCluster.Tests/CodeGenTests/IRuntimeCodeGenGrain.cs

```diff
@@ -33,6 +33,21 @@ public Task M(string arg)
 
     // End regression test for https://github.com/dotnet/orleans/issues/8991
 
+    public interface IGrainWithNonPublicMethods : IGrainWithGuidKey
+    {
+        internal class P1;
+        internal Task M1(P1 arg);
+
+        //protected class P2;
+        //protected Task M2(P2 arg);
```

Comment:
This PR does not fix codegen for `protected` members.

For those I think we should change the body to "throw unreachable" and skip the invoker completely..

Reply:
That makes sense. Good idea